papermillhub/launcher.py

```diff
@@ -0,0 +1,131 @@
+"""
```

Comment:
This is modified from a similar module in `binderhub`, which manages launching user servers, and making requests to the hub API as those users. It could probably be cleaned up a bit, but does the job for now.